components/SpinButton.vue

```diff
@@ -1,124 +1,116 @@
 <template>
-  <div class="d-inline-block">
-    <b-button
-      :variant="variant"
-      :disabled="disabled"
-      :size="size"
-      :class="buttonClass"
-      @click="click"
-    >
-      <span v-if="iconlast">
-        {{ label }}
-      </span>
-      <span>
-        <span v-if="name">
-          <v-icon v-if="done" icon="check" :class="spinclass + ' fa-fw'" />
-          <v-icon
-            v-else-if="doing"
-            icon="sync"
-            :class="'fa-fw fa-spin ' + spinclass"
-          />
-          <v-icon v-else class="fa-fw" :icon="name" />&nbsp;
-        </span>
-        <span v-if="!iconlast" class="ml-1">
-          {{ label }}
-        </span>
-      </span>
-    </b-button>
-    <ConfirmModal
-      v-if="confirm && showConfirm"
-      @confirm="doIt"
-      @hidden="showConfirm = false"
+  <b-button
+    :variant="variant"
+    :disabled="disabled"
+    :size="size"
+    :tabindex="tabindex"
+    :title="buttonTitle"
+    :class="[
+      'd-flex gap-1 align-items-center',
+      noBorder && 'no-border',
+      iconlast && 'flex-row-reverse',
+    ]"
+    @click="onClick"
+  >
+    <v-icon
+      v-if="doing"
+      icon="sync"
+      :class="['fa-spin fa-fw', spinColorClass]"
     />
-  </div>
+    <v-icon v-else-if="done && doneIcon" :icon="doneIcon" :class="iconClass" />
+    <v-icon v-else-if="iconName" :class="iconClass" :icon="iconName" />
+    <span v-if="label || $slots.default">
+      <slot>{{ label }}</slot>
+    </span>
+  </b-button>
 </template>
 <script setup>
-import { ref, defineAsyncComponent } from '#imports'
-
-const ConfirmModal = defineAsyncComponent(() => import('./ConfirmModal'))
+import { ref } from '#imports'
 
 const props = defineProps({
   variant: {
     type: String,
     required: true,
   },
-  name: {
+  iconName: {
     type: String,
     required: false,
     default: null,
   },
   label: {
     type: String,
-    required: true,
+    required: false,
+    default: '',
   },
   timeout: {
     type: Number,
     required: false,
     default: 5000,
   },
-  spinclass: {
+  spinColor: {
```

Comment:
Added the Object with config, should be good. I'd left the prop though just in case it'd be needed in some scenario to override configured color of spinner.